pollux/lib/core/src/main/scala/io/iohk/atala/pollux/core/service/CredentialServiceImpl.scala

```diff
@@ -600,34 +605,48 @@ private class CredentialServiceImpl(
   override def receiveCredentialIssue(
       issueCredential: IssueCredential
   ): ZIO[WalletAccessContext, CredentialServiceError, IssueCredentialRecord] = {
-    // TODO We can get rid of this 'raw' representation stored in DB, because it is not used.
-    val rawIssuedCredential = issueCredential.attachments.map(_.data.asJson.noSpaces).headOption.getOrElse("???")
     for {
       // TODO Move this type of generic/reusable code to a helper trait
-      attachmentFormatAndData <- ZIO.succeed {
-        import IssueCredentialIssuedFormat.{Anoncred, JWT}
-        issueCredential.attachments
-          .collectFirst {
-            case AttachmentDescriptor(_, _, Base64(v), Some(JWT.name), _, _, _, _)      => (JWT, v)
-            case AttachmentDescriptor(_, _, Base64(v), Some(Anoncred.name), _, _, _, _) => (Anoncred, v)
-          }
-          .map { case (f, v) => (f, java.util.Base64.getUrlDecoder.decode(v)) }
-      }
       record <- getRecordFromThreadIdWithState(
         issueCredential.thid.map(DidCommID(_)),
         ignoreWithZeroRetries = true,
         ProtocolState.RequestPending,
         ProtocolState.RequestSent
       )
-      _ <- attachmentFormatAndData match
-        case Some(IssueCredentialIssuedFormat.JWT, _)       => ZIO.succeed(())
-        case Some(IssueCredentialIssuedFormat.Anoncred, ba) => processAnonCredsCredential(record, ba)
-        case _ => ZIO.fail(UnexpectedError("No AnonCreds or JWT credential attachment found"))
+      processedAttachments <- {
+        import IssueCredentialIssuedFormat.Anoncred
+        ZIO.collectAll(
+          issueCredential.attachments
+            .map {
+              case AttachmentDescriptor(
+                    id,
+                    media_type,
+                    Base64(v),
+                    Some(Anoncred.name),
+                    _,
+                    _,
+                    _,
+                    _
+                  ) =>
+                processAnonCredsCredential(record, java.util.Base64.getUrlDecoder.decode(v))
+                  .map(processedCredential =>
+                    AttachmentDescriptor.buildBase64Attachment(
+                      id = id,
+                      mediaType = media_type,
+                      format = Some(IssueCredentialIssuedFormat.Anoncred.name),
+                      payload = processedCredential
+                    )
+                  )
+              case attachment => ZIO.succeed(attachment)
+            }
+        )
+      }
+      processedIssuedCredential = issueCredential.copy(attachments = processedAttachments)
       _ <- credentialRepository
         .updateWithIssuedRawCredential(
           record.id,
-          issueCredential,
-          rawIssuedCredential,
+          processedIssuedCredential,
+          processedIssuedCredential.attachments.map(_.data.asJson.noSpaces).headOption.getOrElse("???"),
```

Comment:
I think it would be nice to figure out what to do when headOption is None
otherwise, we might see some strange `???` in the database.